purchase_sale_link/models/sale_order.py

```diff
@@ -0,0 +1,92 @@
+# © 2019 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models, api, fields
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    purchase_order_count = fields.Integer(
+        string="Number of Purchase Order",
+        compute='_compute_purchase_order_count',
+        groups='purchase.group_purchase_user',
+    )
+
+    @api.multi
+    def get_purchase_orders_service(self):
+        """Get the Purchase Orders linked to one Sale Order
+        by service products purchased automatically"""
+        self.ensure_one()
+
+        purchase_order_lines_service = self.env['purchase.order.line']\
+            .search([('sale_order_id', 'in', self.ids)])
+
+        purchase_orders_service = \
+            purchase_order_lines_service.mapped('order_id')
+
+        return purchase_orders_service
+
+    @api.multi
+    def get_purchase_orders_stock(self):
+        """Get the Purchase Orders linked to one Sale Order
+        by storable products purchased by 'Make To Order' route"""
+        self.ensure_one()
+
+        # Get the related stock moves via the SO's procurement_group_id
+        procurement_group_id = self.procurement_group_id.id
+        stock_moves = self.env['stock.move'].search([
+            ('group_id', '=', procurement_group_id),
+            ('created_purchase_line_id', '!=', 'False'),
+        ])
+
+        purchase_order_lines_stock = stock_moves\
+            .mapped('created_purchase_line_id')
+        purchase_orders_stock = purchase_order_lines_stock.mapped('order_id')
+
+        return purchase_orders_stock
+
+    @api.multi
+    @api.depends('order_line.purchase_line_ids')
+    def _compute_purchase_order_count(self):
+        """
+        Overwrite the original _compute_purchase_order_count() method in
+        sale_purchase/models/sale_order.py
+
+        The purchase_orders created from a service product are catched
+        in this module in a diferent way from the original module.
+        In this actual way, it allows to easily count and show purchase_orders
+        from both storable and service products
+        """
+        for order in self:
+            if order.state not in ('draft', 'sent'):
```

Comment:
Why do we exclude 'draft' and 'sent', when a sale order is cancelled and set back to draft, but the purchase is not yet removed it should remain visible? 

Or if we decide to not include them should there not be a 'else: purchase_order_count = 0' line?